SimG4CMS/Muon/src/MuonSensitiveDetector.cc

```diff
@@ -56,14 +57,17 @@ MuonSensitiveDetector::MuonSensitiveDetector(const std::string& name,
   //
   // Here simply create 1 MuonSlaveSD for the moment
   //
-  LogDebug("MuonSimDebug") << "create MuonSubDetector " << name;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("MuonSim") << "create MuonSubDetector " << name;
+#endif
   detector = new MuonSubDetector(name);
 
-  edm::ESTransientHandle<DDCompactView> cpv;
-  es.get<IdealGeometryRecord>().get(cpv);
-
   //The constants take time to calculate and are needed by many helpers
-  MuonDDDConstants constants(*cpv);
+  edm::ESHandle<MuonGeometryConstants> mdc;
+  es.get<IdealGeometryRecord>().get(mdc);
+  if (!mdc.isValid())
+    throw cms::Exception("Unknown", "MuonSensitiveDetector") << "Cannot find MuonGeometryConstants\n";
```

Comment:
"Unknown" category is unusual, probably just using "MuonSensitiveDetector" as the category is better

Reply:
Done